src/functions/certGen.ts

```
@@ -23,18 +23,22 @@ const certGen: Handler = async (event: any, context?: Context, callback?: Callba
 
     event.Records.forEach(async (record: any) => {
         const testResult: any = JSON.parse(record.body);
-        const generatedCertificateResponse: Promise<ManagedUpload.SendData> = certificateGenerationService.generateCertificate(testResult)
-        .then((response: IGeneratedCertificateResponse) => {
-            return certificateUploadService.uploadCertificate(response);
-        });
+        if (testResult.testResultId.match("\\b[a-zA-Z0-9]{8}\\b-\\b[a-zA-Z0-9]{4}\\b-\\b[a-zA-Z0-9]{4}\\b-\\b[a-zA-Z0-9]{4}\\b-\\b[a-zA-Z0-9]{12}\\b")) {
+            const generatedCertificateResponse: Promise<ManagedUpload.SendData> = certificateGenerationService.generateCertificate(testResult)
+                .then((response: IGeneratedCertificateResponse) => {
+                    return certificateUploadService.uploadCertificate(response);
+                });
 
-        certificateUploadPromises.push(generatedCertificateResponse);
+            certificateUploadPromises.push(generatedCertificateResponse);
+        } else {
+            certificateUploadPromises.push(Promise.reject(new Error("Record does not have valid testResultId for certificate generation")));
+        }
     });
 
     return Promise.all(certificateUploadPromises)
-    .catch((error: AWSError) => {
-        console.error(error);
-    });
+        .catch((error: Error) => {
+            console.error(error);
```

Should we throw the errors the same as I did on ATF Report?

---

done